lib/leanplum_api/faraday_middleware/response_validation.rb

```diff
@@ -27,17 +29,21 @@ def call(environment)
 
     def validate_request_success(success_indicators, requests)
       if requests && success_indicators.size != requests.size
-        fail BadResponseError, "Attempted #{requests.size} operations but received confirmation for #{success_indicators.size}!"
+        fail BadResponseError, "Attempted #{requests.size} operations; responses for only #{success_indicators.size}!"
       end
 
       failures = success_indicators.map.with_index do |indicator, i|
         if indicator[WARN]
           LeanplumApi.configuration.logger.warn((requests ? "Warning for #{requests[i]}: " : '') + indicator[WARN].to_s)
         end
 
-        next nil if indicator[SUCCESS].to_s == 'true'
-
-        requests ? { operation: requests[i], error: indicator } : { error: indicator }
+        if indicator[SUCCESS].to_s == 'true'
+          nil
+        else
+          failure = { message: indicator.key?(SUCCESS) ? indicator.to_s : "No :success key found in #{indicator}" }
+          failure.merge!(operation: requests[i]) if requests
```

Comment:
this will already return the whole hash, so no need for the next line, but, this is nitpicking as well...

Reply:
unfortunately it won't return the whole hash if `requests.nil?`, which happens in a lot of operations

Reply:
oh yeah, you are right!